libraries/core_libs/consensus/src/final_chain/final_chain.cpp

```diff
@@ -116,6 +116,7 @@ class FinalChainImpl final : public FinalChain {
       for (uint64_t num = start; num <= *last_blk_num; ++num) {
         block_headers_cache_.get(num);
       }
+      rewards_.populateDagBlockAuthorLevels(*last_blk_num);
```

Comment:
Why is this called from final_chain?

Reply:
In final chain finalize method is where this is actually used and it requires last block number as a parameter so final chain constructor where we set the last block number seems to be the best place.